src/v/ssx/semaphore.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/sstring.hh>
+
+#include <utility>
+
+namespace ssx {
```

Comment:
Is it worth adding

```c++
using semaphore = seastar::named_semaphore;
```
Weird, this seems to be what's on your branch. What's up with github?